subsys/bluetooth/controller/multithreading_lock.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "multithreading_lock.h"
+
+/* Mutexes are not to be used within ISRs, therefore we use a semaphore with a
```

Comment:
The comment was placed there specifically to address that mutexes are documented in the Zephyr API to not be used with ISRs, while semaphore's are not. @pizi-nordic, what do you prefer, documentation or no documentation?

Reply:
nothing that sleeps can be used within a ISR and both sem_take and mutex_lock can sleep unless asked not to (with K_NO_WAIT)

You can add a comment in `multithreading_lock.h` next to `MULTITHREADING_LOCK_ACQUIRE_FOREVER_WAIT` and `MULTITHREADING_LOCK_ACQUIRE` (btw - why do we have these two macros if they do the same) that these cannot be used inside ISR. This would be true and maybe useful.

Reply:
The two macros indicate a different meaning. To me, it's "accidental" that they result in the same code. I'd like to keep both, in case we change the underlying implementation.
I've removed the comment, but I'm not sure what you want me to add in multithreading_lock.h

Reply:
I don't mind if you add nothing if you prefer. You can add information that sleeping calls cannot be used in ISR but these macros will be used by us only so I don't think it is really needed.